### PR TITLE
(maint) Update Bolt dependency gems

### PR DESCRIPTION
**Pull request by lucywyman**
Updates gem versions to their latest resolved versions for dependencies
of Bolt and pe-bolt-server.

Verified this builds + runs for both Bolt and pe-bolt-server packages.

### CODE REVIEW

**Review by beechtom — APPROVED** [review]

File: configs/components/rubygem-puppet.rb [hunk 1]
```diff
@@ -4,8 +4,8 @@
   pkg.version version
 
   case version
-  when '7.7.0'
-    pkg.md5sum '81cfcab6f6998bf7cfbcf0c91a72a469'
+  when '7.10.0'
+    pkg.md5sum '801e1945b1c483d1d5a4cb9b1caf7578'
   when '6.23.0'
```
Comment (donoghuc) [inline configs/components/rubygem-puppet.rb:9]: It would probably be good to get on latest 6.x puppet for PE.

Comment (lucywyman) [reply configs/components/rubygem-puppet.rb]: d-d-d-done

File: configs/components/rubygem-puppet.rb [hunk 2]
```diff
@@ -4,10 +4,10 @@
   pkg.version version
 
   case version
-  when '7.7.0'
-    pkg.md5sum '81cfcab6f6998bf7cfbcf0c91a72a469'
-  when '6.23.0'
-    pkg.md5sum '10dbdce584a286bed95835420e2fba48'
+  when '7.10.0'
+    pkg.md5sum '801e1945b1c483d1d5a4cb9b1caf7578'
+  when '6.24.0'
```
Comment (beechtom) [inline configs/components/rubygem-puppet.rb:9]: Default version needs to be updated on line 3.

**Comment by donoghuc** [comment]
I'm sure these are all mirrored in artifactory right?

**Comment by lucywyman** [comment]
Yeah, mirroring is how I get the md5 sums for exactly that reason (so I know they're mirrored). I also got a successful builds of puppet-bolt and pe-bolt-server using these changes in the runtime.

**Review by donoghuc — APPROVED** [review]

**Pull request merged**